sdk/spring/spring-messaging-azure-servicebus/src/main/java/com/azure/spring/messaging/servicebus/core/DefaultServiceBusNamespaceConsumerFactory.java

```diff
@@ -0,0 +1,214 @@
+     */
+    public DefaultServiceBusNamespaceConsumerFactory(NamespaceProperties namespaceProperties) {
+        this(namespaceProperties, key -> null);
+    }
+
+    /**
+     * Construct a factory with the provided namespace level configuration and producer {@link PropertiesSupplier}.
+     * @param namespaceProperties the namespace properties.
+     * @param supplier the {@link PropertiesSupplier} to supply {@link ConsumerProperties} for each queue/topic entity.
+     */
+    public DefaultServiceBusNamespaceConsumerFactory(NamespaceProperties namespaceProperties,
+                                                     PropertiesSupplier<ConsumerIdentifier, ConsumerProperties> supplier) {
+        this.namespaceProperties = namespaceProperties;
+        this.propertiesSupplier = supplier == null ? key -> null : supplier;
+    }
+
+    @Override
+    public ServiceBusSessionReceiverClient createReceiver(String name) {
+        return doCreateReceiver(name, null);
+    }
+
+    @Override
+    public ServiceBusSessionReceiverClient createReceiver(String name, ServiceBusEntityType entityType) {
+        ConsumerProperties consumerProperties = this.propertiesSupplier.getProperties(new ConsumerIdentifier(name)) != null
+            ? this.propertiesSupplier.getProperties(new ConsumerIdentifier(name)) : new ConsumerProperties();
+        if (entityType != null) {
+            consumerProperties.setEntityType(entityType);
+        }
+        return doCreateReceiver(name, consumerProperties);
+    }
+
+    @Override
+    public void addListener(Listener listener) {
+        this.listeners.add(listener);
+    }
+
+    @Override
+    public boolean removeListener(Listener listener) {
+        return this.listeners.remove(listener);
+    }
+
+    @Override
+    public void destroy() {
+        clients.forEach((name, receiver) -> {
+            listeners.forEach(l -> l.receiverRemoved(name, receiver));
+            receiver.close();
+        });
+        this.clients.clear();
+        this.listeners.clear();
+    }
+
+    private ServiceBusSessionReceiverClient doCreateReceiver(String name, @Nullable ConsumerProperties properties) {
+        return clients.computeIfAbsent(name, entityName -> {
+            ConsumerProperties consumerProperties = parentMerger.merge(properties, this.namespaceProperties);
+            consumerProperties.setEntityName(entityName);
+
+            ServiceBusSessionReceiverClient receiverClient;
+            if (Boolean.TRUE.equals(consumerProperties.getSessionEnabled())) {
+                ServiceBusSessionReceiverClientBuilderFactory factory =
+                    new ServiceBusSessionReceiverClientBuilderFactory(consumerProperties, this.customizers);
+
+                factory.setDefaultTokenCredential(this.defaultCredential);
+                factory.setTokenCredentialResolver(this.tokenCredentialResolver);
+                factory.setSpringIdentifier(AzureSpringIdentifier.AZURE_SPRING_INTEGRATION_SERVICE_BUS);
```

Comment:
is there a messaging identifier now?

Reply:
No, the message and integration layers are not yet distinguished.